ragna/deploy/_ui/modal_configuration.py

```diff
@@ -115,12 +128,34 @@ def __init__(self, api_wrapper, **params):
 
         self.got_timezone = False
 
-    def did_click_on_start_chat_button(self, event):
-        if not self.document_uploader.can_proceed_to_upload():
-            self.change_upload_files_label("missing_file")
-        else:
+        self.corpus_or_upload_radiobutton = pn.widgets.RadioButtonGroup.from_param(
+            self.param.corpus_or_upload,
+            button_style="outline",
+            button_type="primary",
+        )
+
+        # self.metadata_filters_builder = MetadataFiltersBuilder()
+
+        self.create_config(components)
```

Comment:
There was a race-condition in the old related to the fact that `model_section` (which contained the logic for `create_config`) was an async function. Depending on how `model_section` was scheduled, the config could actually end up being created after a bunch of functions which depended on it, resulting in a blank modal.

I've simplified things by extracting the logic into a single synchronous function. There's nothing expensive in this function, and so it should be fine to do this. It has the benefit of removing the race-condition and making the code easier to reason about.